src/error.ts

```diff
@@ -136,23 +135,34 @@ export class MongoServerError extends MongoError {
   code?: number;
   codeName?: string;
   writeConcernError?: Document;
+  ok?: number;
+  topologyVersion?: TopologyVersion;
+  [key: string]: any;
 
-  constructor(message: Error | ErrorDescription) {
-    if (message instanceof Error) {
-      super(message);
-    } else {
-      super(message.message || message.errmsg || message.$err || 'n/a');
-      if (message.errorLabels) {
-        this[kErrorLabels] = new Set(message.errorLabels);
-      }
-
-      for (const name in message) {
-        if (name === 'errorLabels' || name === 'errmsg' || name === 'message') {
-          continue;
-        }
-
-        (this as any)[name] = message[name];
-      }
+  constructor(message: ErrorDescription) {
+    super(message.message || message.errmsg || message.$err || 'n/a');
+    if (message.errorLabels) {
+      this[kErrorLabels] = new Set(message.errorLabels);
+    }
+
+    this.code = message.code;
+    this.codeName = message.codeName;
+    this.writeConcernError = message.writeConcernError;
+    this.ok = message.ok;
+    this.topologyVersion = message.topologyVersion;
+
+    const currentProps = Object.getOwnPropertyNames(this);
```

Comment:
Is all this really necessary to simplify the multiple constructor calls in bulk/common? It looks like all this filtering is doing the same thing as the original for..in, but it's less efficient because of the multiple array function calls. Ideally, the server error class should not need to be touched at all to complete this ticket, and we definitely should not be adding new functionality to it unless absolutely necessary (referring to the kErrorLabels line above), so any logic that's specific to particular errors, and not all server errors, should live in those specific classes.

Reply:
Neal and I talked things over and I removed the tons of array calls that were there. I opted to put prop names into a set for better efficiency and it'll likely be easier to read/maintain than loads of if conditions.